### DISCUSSION TITLE
Binding symbols to their docstring counterparts

### DISCUSSION THREAD

**Discussion by sudoandros** [microsoft/pyright#7244] [Ideas]
Hi! First of all, great tool, thank you! But I have one problem with it, maybe you'll be able to fix it. I'd like for code symbols to be "bound" to their description in the docstring. The primary use case I'm thinking of is renaming function argument or class property in the docstring when I'm renaming it in the code. For instance, I have the function `is_prime` like this:
```python
def is_prime(n: int):
    """Check if a number is prime.

    Args:
        n (int): The integer to check for primality.

    Returns:
        bool: True if the number is prime, False otherwise.
    """
    if n <  2:
        return False
    if n ==  2:
        return True
    if n %  2 ==  0:
        return False

    sqrt_n = int(math.floor(math.sqrt(n)))
    return all(n % i !=  0 for i in range(3, sqrt_n +  1,  2))
```

When I'm renaming the argument `n` to `x` or anything else, the name in the `n (int): The integer to check for primality.` string should be changed too. Given that you've already implemented some docstring parsing (#292), I imagine it's not very hard to implement? I don't know a lot about the pyright internals so I may be mistaken. Is there any reason this feature hasn't been implemented? 

AFAIK PyCharm has this feature and it's the only real disadvantage I see in your project comparing to it. It would be a great addition to pyright which allows developers to better avoid mistakes in the project's API documentation.

**Comment by erictraut** [comment]
Pyright is a static type checker. While it does have some basic language server functionality like simple semantic-aware renaming, we don't plan to add new language server features like the one you're suggesting here.

If you are interested in more advanced language server features, consider switching to Pylance, Microsoft's Python language server built on top of pyright. Pylance is free and is licensed for use in VS Code. It has a full team of developers supporting it. If you'd like to suggest new language server features for Pylance, use [its issue tracker](https://github.com/microsoft/pylance-release).

**Reply by sudoandros** [reply]
Oh, I see. Sorry to disturb you, thank you for the response :)
